File: Lista_8/lista8_ex9.c
/* Exercício 9:
Escreva uma função recursiva que calcule [lg n], ou seja, o piso do logaritmo de n na base 2.
*/
#include <stdio.h>

int recLog2n(int n);

int main() {
int n;

printf("Digite o valor de n, para obter x, de 2^x = n: ");
scanf("%d", &n);

if(n < 1) {
    printf("\nErro: Valores devem ser inteiros positivos!\n");
} else {
    printf("\nResultado: %d\n", recLog2n(n));
}
return 0;
}

int recLog2n(int n) {
    if(n == 1) {
        return 0;
    }
    return 1 + recLog2n(n/2);
}

File: Lista_8/Lista8_ex9.py
def log_recursivo(n):
    if n == 1:
        return 0
    else:
        return 1 + log_recursivo(n/2)

def main():
    n = int(input("Digite o valor de n, para obter x, de 2^x = n: "))

    if n < 1:
        print("\nErro: Valores devem ser inteiros positivos!\n")
    else:
        print("Resultado: %d\n" % log_recursivo(n))

if __name__ == "__main__":
	main()
